babel/messages/pofile.py

```diff
@@ -142,7 +144,8 @@ def __init__(self, *args: str) -> None:
             self.append(arg)
 
     def append(self, s: str) -> None:
-        self._strs.append(s.strip())
+        assert s[0] == '"' and s[-1] == '"'
```

Comment:
Since this looks to be a hard constraint for `_NormalizedString` to denormalize correctly (since `unescape` always strips the first and last character), this should probably not be an `assert` that could get stripped out by `-O` modes.

```suggestion
        if not (s and s[0] == '"' and s[-1] == '"'):
             raise ValueError(f"_NormalizedString components must be double quoted; got {s!r}")
```
maybe?

Reply:
I guess that depends on how you want to treat this class. Since this is a non-user-facing class, I thought I'd model this requirement as an assertion the code does on its input, and that babel developers shouldn't break -- users should never see this failing, assertions enabled or disabled. In other words, if users were to see such an exception, there'd be nothing they could do about it.

Taking one step behind, I think ideally I'd rather see this class take strings with the double quotes already removed in the first place, which is way less confusing. However I think I saw the `unescape` function being used somewhere else, and I didn't want to introduce breaking behaviour. I'm also not sure whether the function is part of babel's public API (judging by the name it might be).

Reply:
Leaving as-is at the moment, let me know if you want this to change.